core/src/ephemeral_server/mod.rs

```diff
@@ -372,14 +372,54 @@ impl EphemeralExe {
     }
 }
 
+/// Returns a TCP port that is available to listen on for the given local host.
+///
+/// This works by binding a new TCP socket on port 0, which requests the OS to
+/// allocate a free port. There is no strict guarantee that the port will remain
+/// available after this function returns, but it should be safe to assume that
+/// a given port will not be allocated again to any process on this machine
+/// within a few seconds.
+///
+/// On Unix-based systems, binding to the port returned by this function
+/// requires setting the `SO_REUSEADDR` socket option (Rust already does that by
+/// default, but other languages may not); otherwise, the OS may fail with a
+/// message such as "address already in use". Windows default behavior is
+/// already appropriate in this regard; on that platform, `SO_REUSEADDR` has a
+/// different meaning and should not be set (setting it may have unpredictable
+/// consequences).
+#[allow(clippy::let_and_return)]
 fn get_free_port(bind_ip: &str) -> u16 {
-    // Can just ask OS to give us a port then close socket. OS's don't give that
-    // port back to anyone else anytime soon.
-    std::net::TcpListener::bind(format!("{bind_ip}:0"))
-        .unwrap()
-        .local_addr()
-        .unwrap()
-        .port()
+    let listen = std::net::TcpListener::bind((bind_ip, 0)).unwrap();
```

Comment:
IMO no reason not to make this return a result and unwrap it in places that don't care about erroring.

Reply:
Sure, will fix.